provider/resource_wireguard_preshared_key.go

```diff
@@ -0,0 +1,55 @@
+package provider
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+	"io"
+)
+
+func resourceWireguardPresharedKey() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provides a WireGuard asymmetric key resource. This can be used to create, read, and delete WireGuard keys in terraform state.",
+
+		Create: resourceWireguardPresharedKeyCreate,
+		Read:   resourceWireguardPresharedKeyRead,
+		Delete: resourceWireguardPresharedKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"preshared_key": {
+				Description: "Additional layer of symmetric-key cryptography to be mixed into the already existing public-key cryptography, for post-quantum resistance.",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceWireguardPresharedKeyCreate(d *schema.ResourceData, m interface{}) error {
+	var key wgtypes.Key
+	var err error
+
+	key, err = wgtypes.GenerateKey()
```

Comment:
I don't write much Go, why like this vs.

```suggestion
	key, err := wgtypes.GenerateKey()
```

? As far as I understand `var` declares a mutable variable, but `key` is only assigned once.

Reply:
Using `key, err := wgtypes.GenerateKey()` would make the declaration of the variables in 30, 31 obsolete.

Reply:
Right, but they're not necessary? Especially `key` as it's only assigned once. Not that bothered, was just curious.

Reply:
I think both is syntactically ok. `:=` is a declaration, whereas `=` is an assignment. But it doesn't matter that much inside the function block.